typing/types.ml

```diff
@@ -267,10 +267,10 @@ and signature = signature_item list
 
 and signature_item =
     Sig_value of Ident.t * value_description
-  | Sig_type of Ident.t * type_declaration * rec_status
+  | Sig_type of Ident.t * type_declaration * rec_status * private_flag
```

Comment:
I think adding a separate boolean type for this would be better than using `Asttypes.private_flag`.